### ISSUE TITLE
Critical vulnerability in 3.13:libjpeg-turbo (alpine 3.13).

### ISSUE THREAD

**Issue by rahulsadanandan** [libjpeg-turbo/libjpeg-turbo#514]
**Have you searched the existing issues (both open and closed) in the libjpeg-turbo issue tracker to ensure that this bug report is not a duplicate?**
Yes

On scanning Nginx docker image version(image tag): 1.19.10-alpine I have noticed the below Vulnerabilitity tagged with Critical Severity.

```
{
   "user_issue_id":"1382170090968199168",
   "comp_id":"build://artifactory-pro-installers-docker-release:7.19.0-m009-20210414012920",
   "comp_name":"artifactory-pro-installers-docker-release",
   "comp_version":"7.19.0-m009-20210414012920",
   "type":"security",
   "targets":null,
   "issue_id":"XRAY-171859",
   "paths":[
      "default/builds/artifactory-pro-installers-docker-release"
   ],
   "summary":"libjpeg-turbo rdppm.c get_word_rgb_row() Function 16-bit PPM / PGM File Handling Heap Buffer Overflow",
   "severity":"Critical",
   "updated":"2021-04-14T03:13:23Z",
   "permissions":{
      
   },
   "is_source_root":false,
   "source":"3.13:libjpeg-turbo",
   "source_version":"2.0.6-r0",
   "source_id":"alpine://3.13:libjpeg-turbo"
}
```

It seems like `3.13:libjpeg-turbo` version - `2.0.6-r0` is the affected component. I saw that a patch was merged to master some days back
https://github.com/libjpeg-turbo/libjpeg-turbo/commit/f35fd27ec641c42d6b115bfa595e483ec58188d2

@dcommander wanted to know when this released as this is a critical vulnerability.
cc @thresheek
Thanks

**Comment by dcommander** [comment]
I'm not even sure which issue that JSON file is referring to.  The commit you referenced above was a bug fix for an issue I just discovered last week while developing new libjpeg-turbo fuzz targets for Google's OSS-Fuzz platform.  The commit message explains why it wasn't a critical vulnerability.  In order for software to expose that issue to an end user, the software would have had to basically force the `tjLoadImage()` function to load an RGB PPM file into a grayscale image buffer.  In order for a bug to be exploitable from a security point of view, it has to be a bug in a code path that causes that code path to behave badly with specific inputs while behaving normally for most inputs.  In this case, the code path would have crashed under any circumstances, because it was basically the same as doing:

```
unsigned char *grayscale, *rgb;

rgb = (unsigned char *)malloc(numPixels * 3);
grayscale = (unsigned char *)malloc(numPixels);
memcpy(grayscale, rgb, numPixels * 3);
```

The only way such a glaring issue would be a potential security vulnerability is if an application developer let such a bug slip into the final release of their application, which is extremely unlikely.  Even the TurboJPEG test programs (TJBench, TJExample) could not reproduce the issue, because they never try to force-load RGB files into grayscale buffers.  Thus, the bug fix wasn't so much fixing a known vulnerability as it was user-proofing the API.

**Comment by dcommander** [comment]
But to answer your question, if that is in fact the issue that the JSON file is referring to, the fix will be released in 2.1, which should be available within the next couple of weeks.  The fix has also been back-ported to the 2.0.x Git branch, which has all post-2.0.6 bug fixes.  There will not be a 2.0.7 release, since the 2.1 branch is stable now.  I will however, continue to back-port relevant bug fixes into the 2.0.x branch as long as the 2.1 branch is the active branch.  (Generally speaking, I  maintain the current branch and one older branch, and I'll back-port critical bug fixes into even older branches as well as long as the fixes apply cleanly.  I also provide continuing support for older code branches on a paid basis.)

**Comment by rahulsadanandan** [comment]
@dcommander Thanks very much for the detailed update.
